sdk/cosmosdb/cosmos/src/request/request.ts

```diff
@@ -128,6 +129,13 @@ export async function getHeaders({
   }
 
   if (options.maxIntegratedCacheStalenessInMs && resourceType === ResourceType.item) {
+    if (options.maxIntegratedCacheStalenessInMs === 0) {
+      // TODO - log error with Cosmos diagnosics.
+      logger.error(
+        `maxIntegratedCacheStalenessInMs cannot be zero. Default value is null Cache Staleness is supported in milliseconds granularity. Anything smaller than milliseconds or 0 will be ignored. value: ${options.maxIntegratedCacheStalenessInMs}`
```

Comment:
_maxIntegratedCacheStalenessInMs cannot be zero. Default value is null Cache Staleness is supported in milliseconds granularity. Anything smaller than milliseconds or 0 will be ignored. value_

add 'and' or ',' in the sentence. 
maxIntegratedCacheStalenessInMs cannot be zero. Default value is null and Cache Staleness is supported in milliseconds granularity. Anything smaller than milliseconds or 0 will be ignored.

Reply:
0 should be a valid value for maxIntegratedCacheStalenessInMs as documented here: https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-integrated-cache#adjust-maxintegratedcachestaleness

Effectively, this means that the cache should be bypassed for this request.

Reply:
@jcocchi we don't need this PR. the documentation says:  
https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-integrated-cache#adjust-maxintegratedcachestaleness

maxIntegratedCacheStalenessInMs = 0
If you set MaxIntegratedCacheStaleness to 0, your read request will never use the integrated cache, regardless of the consistency level. 

maxIntegratedCacheStalenessInMs = null or not present in the request header.
When not configured, the default MaxIntegratedCacheStaleness is 5 minutes.

@kirankumarkolli does this read the same for you?

Reply:
maxIntegratedCacheStaleness should be able to be set to 0, in which case the gateway will take care of bypassing the cache. We shouldn't throw an error when this is set to 0.

The current behavior is if I set this to 0 then the default staleness of 5 minutes is applied so the ask was to change that to allow a staleness of 0.